pkg/templates/notificationtemplates/notification_generator.go

```diff
@@ -21,31 +29,41 @@ type NotificationTemplate struct {
 	Name    string
 }
 
-// GetNotificationTemplate returns a notification subject, body and a boolean
-// indicating whether or not a template was found. Otherwise, an error will be returned
-func GetNotificationTemplate(name string) (*NotificationTemplate, bool, error) {
-	templates, err := loadTemplates()
+// GetNotificationTemplate returns a NotificationTemplate with the given name for the specified notificationTemplateSetName. An error will be returned if such a template is not found or there is an error in loading templates.
+// This function expects the templates to be organized under rootDirectory as rootDirectory/notificationTemplateSetName/templateName/notification.html and rootDirectory/notificationTemplateSetName/templateName/subject.txt
+// making the function dependent on the path length.
+func GetNotificationTemplate(name string, notificationTemplateSetName string) (*NotificationTemplate, error) {
+	templates, err := loadTemplates(notificationTemplateSetName)
 	if err != nil {
-		return nil, false, errors.Wrap(err, "unable to get notification templates")
+		return nil, errors.Wrap(err, "unable to get notification templates")
 	}
 	template, found := templates[name]
-	return &template, found, nil
+	if !found {
+		return &template, fmt.Errorf("notification template %v not found in %v", name, notificationTemplateSetName)
+	}
+	return &template, nil
 }
 
-func templatesForAssets(assets assets.Assets) (map[string]NotificationTemplate, error) {
-	paths := assets.Names()
+func templatesForAssets(efs embed.FS, root string, env string) (map[string]NotificationTemplate, error) {
```

Comment:
I see you still use `env` as the template set name, right? Can we get it renamed to `setName` maybe?

Reply:
done in aace73a

Reply:
oops, i thought i fixed that already. done in [29be8b4](https://github.com/codeready-toolchain/host-operator/pull/754/commits/29be8b41347898468009c261876303dde8b37305) now